src/Cli/dotnet/commands/dotnet-workload/install/WorkloadManifestUpdater.cs

```diff
@@ -88,10 +88,10 @@ public async Task UpdateAdvertisingManifestsAsync(bool includePreviews, bool use
             }
         }
 
-        public async void DownloadWorkloadSet(string version, DirectoryPath? offlineCache = null)
+        public void DownloadWorkloadSet(string version, DirectoryPath? offlineCache = null)
         {
             var correctedVersion = WorkloadSetVersionToWorkloadSetPackageVersion(version);
-            await UpdateManifestWithVersionAsync("Microsoft.NET.Workloads", includePreviews: true, _sdkFeatureBand, new NuGetVersion(correctedVersion), offlineCache);
+            Task.Run(() => UpdateManifestWithVersionAsync("Microsoft.NET.Workloads", includePreviews: true, _sdkFeatureBand, new NuGetVersion(correctedVersion), offlineCache)).Wait();
```

Comment:
This definitely shouldn't have been `async void`.  It could be `async Task` instead but then you need to call a wait further up in the call stack.

Is there any reason not to do this instead though?  It seems simpler:

```suggestion
            UpdateManifestWithVersionAsync("Microsoft.NET.Workloads", includePreviews: true, _sdkFeatureBand, new NuGetVersion(correctedVersion), offlineCache).Wait();
```

I'm not an async expert though, maybe running it on another thread with `Task.Run` would help prevent deadlocks (it could also cause them for all I know though).

Reply:
I am definitely not an async expert either, but I was reading this:
https://grantwinney.com/call-an-async-method-from-a-synchronous-one/

and he said that just calling .Wait() afterwards would make it deadlock; putting it in Task.Run() first would mean that the waiting thread and the thread with work are separate, which prevents deadlocks.

This is obviously very weak evidence, but I ran it several times, and it didn't deadlock 🤷

Reply:
To my knowledge, if the program is running on a single thread, then calling `.Wait` without `Task.Run` can deadlock. 
`Wait` is designed to be used for when you want to block the calling thread, without `Task.Run` this could end up being the same thread and cause a deadlock.
See: https://learn.microsoft.com/en-us/dotnet/api/system.threading.tasks.task.wait?view=net-8.0#:~:text=//%20%20%20%20%20%20%20N%3A%20%20%20%20%20%20%201%2C000%2C000-,Remarks,-Wait%20is%20a

Is there a reason you can't use `await` instead? This will simply pause the method and add it back to the threadpool to be executed later, whenever the`awaited` thing finishes. Either way, I think this should be fine.

Reply:
I'm sure you already know this, but never take running it a few times or even a few thousand times as evidence 😉 

It is always worth fully understanding async code before committing it. It's very easy to create silent killers over time that slowly erode at a codebase's reliability and create bug reports that are a real headache. Eventually the bugs will compound and exponentially decay the software's ability to function at all. By that point, enough code will probably be written on top of the deadlocks / race conditions that it will be impossible to decouple, unravel, and synchronize the logic properly, so you'll end up having to dump the whole thing out

Reply:
To use `await`, you have to be in an `async` method, which means the caller (or somewhere up the stack) will need to do something like `Wait()`, or you'll need to make all your methods up the stack to the entry point async.